src/harness/testcases/WINNF_FT_S_HBT_testcase.py

```diff
@@ -0,0 +1,191 @@
+    del request, response
+
+    # First Heartbeat to authorize the device.
+    request = {
+        'heartbeatRequest': [{
+            'cbsdId': cbsd_ids[0],
+            'grantId': grant_id,
+            'operationState': 'GRANTED'
+        }]
+    }
+
+    response = self._sas.Heartbeat(request)['heartbeatResponse'][0]
+    # Check the heartbeat response.
+    self.assertEqual(response['cbsdId'], cbsd_ids[0])
+    self.assertEqual(response['response']['responseCode'], 0)
+    del request, response
+
+    # Calculate (wait time) difference between current time and GrantExpireTime.
+    difference_time = (grant_expire_time - datetime.utcnow()).total_seconds()
+    logging.debug(
+        'Difference between grantExpireTime and CurrentTime (in seconds): %d',
+        difference_time)
+    time.sleep(difference_time + 1)
+
+    # Send a Heartbeat request after grantExpireTime with expired grantId.
+    request = {
+        'heartbeatRequest': [{
+            'cbsdId': cbsd_ids[0],
+            'grantId': grant_id,
+            'operationState': 'GRANTED'
+        }]
+    }
+    response = self._sas.Heartbeat(request)['heartbeatResponse'][0]
+    # Check the heartbeat response, must have Response Code 103
+    self.assertEqual(response['cbsdId'], cbsd_ids[0])
+    self.assertTrue(response['response']['responseCode'] in (103, 500))
+    transmit_expire_time = datetime.strptime(response['transmitExpireTime'],
+                                             '%Y-%m-%dT%H:%M:%SZ')
+    self.assertLessEqual(transmit_expire_time, datetime.utcnow())
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_HBT_8(self):
+    """Heartbeat Array Request: BLACKLISTED.
+
+    Response Code should  be 101.
+    """
+    # Register the devices
+    registration_request = []
+    fcc_ids = []
+    for device_filename in ('device_a.json', 'device_c.json', 'device_e.json'):
+      device = json.load(
+          open(os.path.join('testcases', 'testdata', device_filename)))
+      fcc_ids.append(device['fccId'])
+      registration_request.append(device)
+
+    cbsd_ids = self.assertRegistered(registration_request)
+
+    # Request grant
+    grant_request = []
+    for cbsd_id in cbsd_ids:
+      grant = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+      grant['cbsdId'] = cbsd_id
+      grant_request.append(grant)
+    request = {'grantRequest': grant_request}
+
+    # Check grant response
+    response = self._sas.Grant(request)['grantResponse']
+    self.assertEqual(len(response), len(cbsd_ids))
+    grant_expire_time = []
+    for resp_number, resp in enumerate(response):
+      self.assertEqual(resp['cbsdId'], cbsd_ids[resp_number])
+      self.assertTrue(resp['grantId'])
+      self.assertEqual(resp['response']['responseCode'], 0)
+      grant_expire_time.append(
+          datetime.strptime(resp['grantExpireTime'], '%Y-%m-%dT%H:%M:%SZ'))
+
+    grant_ids = [resp['grantId'] for resp in response]
```

Comment:
I think we don't to do another loop here, we could use the previous one, right?

Reply:
Changed it to use the previous loop.